source/lambda/apiprocessor/redaction.py

```diff
@@ -0,0 +1,140 @@
+
+######################################################################################################################
+#  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                           #
+#                                                                                                                    #
+#  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance    #
+#  with the License. A copy of the License is located at                                                             #
+#                                                                                                                    #
+#      http://www.apache.org/licenses/LICENSE-2.0                                                                    #
+#                                                                                                                    #
+#  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES #
+#  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    #
+#  and limitations under the License.                                                                                #
+#####################################################################################################################
+
+import boto3
+import json
+import os
+from helper import S3Helper
+
+# the maximum number of exclusion lists we support, like legal, marketing, etc
+MAXIMUM_EXCLUSION_LISTS = 100
+
+# get the redacted items of this document
+def getDocumentRedaction(documentId, bucket):
+
+    redaction = {}
+    redaction['uuid'] = documentId
+    redaction['redactedItems'] = []
+    redaction['headers'] = []
+    redaction['footers'] = []
+    
+    # the redaction file may or may not exist
+    try:
+        data = S3Helper.readFromS3(bucket, 'public/' + documentId + '/redaction.json')
+        redaction = json.loads(data)
+    except Exception as e:
+        print("getDocumentRedaction exception: " + str(e))
+    except:
+        print("getDocumentRedaction exception")
+    
+    return redaction
+
+
+# retrieve the redacted items of this document
+def saveDocumentRedaction(documentId, bucket, body):
```

Comment:
Is it a security risk if we don't validate the data of the body?

Reply:
Yep, All API input needs to be validated with the appropriate response codes